Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs

```diff
@@ -0,0 +1,727 @@
+    public static readonly BindableProperty ItemSizingStrategyProperty = BindableProperty.Create(nameof(ItemSizingStrategy), typeof(ItemSizingStrategy), typeof(DataGrid), DeviceInfo.Platform == DevicePlatform.Android ? ItemSizingStrategy.MeasureAllItems : ItemSizingStrategy.MeasureFirstItem);
+    public static readonly BindableProperty ColumnsProperty = BindableProperty.Create(nameof(Columns), typeof(ObservableCollection<DataGridColumn>), typeof(DataGrid), propertyChanged: OnColumnChanged, defaultValueCreator: _ => new ObservableCollection<DataGridColumn>());
+    public static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(DataGrid), null, propertyChanged: OnItemsSourceChanged);
+    public static readonly BindableProperty IsSortableProperty = BindableProperty.Create(nameof(IsSortable), typeof(bool), typeof(DataGrid), false);
+    public static readonly BindableProperty SelectionModeProperty = BindableProperty.Create(nameof(SelectionMode), typeof(SelectionMode), typeof(DataGrid), Microsoft.Maui.Controls.SelectionMode.None, BindingMode.TwoWay, propertyChanged: OnSelectionModeChanged);
+    public static readonly BindableProperty SortIconProperty = BindableProperty.Create(nameof(SortIcon), typeof(Polygon), typeof(DataGrid));
+    public static readonly BindableProperty NoDataViewProperty = BindableProperty.Create(nameof(NoDataView), typeof(View), typeof(DataGrid), propertyChanged: OnNoDataViewChanged);
+    public static readonly BindableProperty ShowDividerProperty = BindableProperty.Create(nameof(ShowDivider), typeof(bool), typeof(DataGrid), true, propertyChanged: OnShowDividerPropertyChanged);
+    public static readonly BindableProperty SelectedItemsProperty = BindableProperty.Create(nameof(SelectedItems), typeof(IList<object>), typeof(DataGrid), null, BindingMode.TwoWay);
+    public static readonly BindableProperty SelectedItemProperty = BindableProperty.Create(nameof(SelectedItem), typeof(object), typeof(DataGrid), null, BindingMode.TwoWay,
+        propertyChanged: (bindable, _, newValue) =>
+        {
+            var self = (DataGrid)bindable;
+            if (self._collectionView.SelectedItem != newValue)
+            {
+                self._collectionView.SelectedItem = newValue;
+            }
+        },
+        coerceValue: (bindable, value) =>
+        {
+            var self = (DataGrid)bindable;
+            if (self.SelectionMode == SelectionMode.None)
+            {
+                return null;
+            }
+
+            return value;
+        }
+    );
+    public static readonly BindableProperty SortedColumnIndexProperty = BindableProperty.Create(nameof(SortedColumnIndex), typeof(SortData), typeof(DataGrid), null, BindingMode.TwoWay,
+        (bindable, value) =>
```

Comment:
Same here, see if you can move these to separate method

Reply:
Done